jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -0,0 +1,250 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.logging;
+
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Consumer;
+
+import org.slf4j.ILoggerFactory;
+import org.slf4j.Logger;
+
+public class JettyLoggerFactory implements ILoggerFactory
+{
+    protected static JettyLoggerFactory getLoggerFactory()
+    {
+        if (instance == null)
+        {
+            instance = new JettyLoggerFactory();
+        }
+
+        return instance;
+    }
+
+    protected static void setInstance(JettyLoggerFactory loggerFactory)
```

Comment:
Why do we need this at all? Just for testing?

Reply:
It's to help with testing, yes.
That's also the reason it's protected and not public.